### PR TITLE
refactor: space between, closes #4256

### LINKED ISSUE #4256: Replace `SpaceBetween` component using Panda
Replace stacks ui based reusable layout

### PR DESCRIPTION
**Pull request by fbwoolf**
> Try out this version of Leather — download [extension builds](https://github.com/leather-wallet/extension/actions/runs/6303759536).

Refactors our shared `SpaceBetween` component to use `HStack`.

### CODE REVIEW

**Review by alter-eggo — APPROVED** [review]

File: src/app/pages/rpc-send-transfer/components/send-transfer-details.tsx [hunk 1]
```diff
@@ -13,21 +12,21 @@ export function SendTransferDetails({ address, amount, currentAddress }: SendTra
   return (
     <Stack
       border="4px solid"
-      borderColor={color('border')}
+      borderColor="accent.border-default !important"
```
Comment (pete-watters) [inline src/app/pages/rpc-send-transfer/components/send-transfer-details.tsx:13]: Do we really need the `important` ?

**Review by pete-watters — APPROVED** [review]

**Review by kyranjamie — APPROVED** [review]
Would it be worth removing `SpaceBetween` component entirely? What does it do that the `Stack` components doesn't already?

File: src/app/pages/rpc-send-transfer/components/send-transfer-details.tsx (hunk 1 above)
Comment (fbwoolf) [reply src/app/pages/rpc-send-transfer/components/send-transfer-details.tsx:13]: I have needed to add this when overriding the border color. If you know a better way to get this to work, let me know.

**Comment by fbwoolf** [comment]
> Would it be worth removing `SpaceBetween` component entirely? What does it do that the `Stack` components doesn't already?

It is an `HStack`. I think someone created it so people didn't have to repeat doing the `isInline` and adding the `space-between` to it. If you want to remove it, I can.

**Pull request merged**